src/main/java/org/zalando/nakadi/domain/NakadiCursor.java

```diff
@@ -0,0 +1,61 @@
+package org.zalando.nakadi.domain;
+
+import java.util.Objects;
+
+public class NakadiCursor {
+    private final String topic;
+    private final String partition;
+    // NO BEGIN OR END HERE!
+    private final String offset;
+
+    public NakadiCursor(final String topic, final String partition, final String offset) {
+        assert null != topic && null != partition && null != offset;
```

Comment:
Why do you check for nulls here? If it's possible that nulls are passed here - please define params as Nullable. If no nulls are expected - then I think it's better to remove checks.